network_analysis/trace-a-utility-network/src/main/java/com/esri/samples/trace_a_utility_network/TraceAUtilityNetworkController.java

```diff
@@ -0,0 +1,493 @@
+    utilityTraceParameters.setTraceConfiguration(mediumVoltageTier.getTraceConfiguration());
+
+    // run the utility trace and get the results
+    ListenableFuture<List<UtilityTraceResult>> utilityTraceResultsFuture =
+        utilityNetwork.traceAsync(utilityTraceParameters);
+    utilityTraceResultsFuture.addDoneListener(() -> {
+      try {
+        List<UtilityTraceResult> utilityTraceResults = utilityTraceResultsFuture.get();
+
+        if (utilityTraceResults.get(0) instanceof UtilityElementTraceResult) {
+          UtilityElementTraceResult utilityElementTraceResult = (UtilityElementTraceResult) utilityTraceResults.get(0);
+
+          if (!utilityElementTraceResult.getElements().isEmpty()) {
+
+            // iterate through the map's feature layers
+            mapView.getMap().getOperationalLayers().forEach(layer -> {
+              if (layer instanceof FeatureLayer) {
+
+                // create query parameters to find features who's network source name matches the layer's feature
+                // table name
+                QueryParameters queryParameters = new QueryParameters();
+                utilityElementTraceResult.getElements().forEach(utilityElement -> {
+
+                  String networkSourceName = utilityElement.getNetworkSource().getName();
+                  String featureTableName = ((FeatureLayer) layer).getFeatureTable().getTableName();
+
+                  if (networkSourceName.equals(featureTableName)) {
+                    queryParameters.getObjectIds().add(utilityElement.getObjectId());
+                  }
+                });
+
+                // select features that match the query
+                ListenableFuture<FeatureQueryResult> featureQueryResultListenableFuture =
+                    ((FeatureLayer) layer).selectFeaturesAsync(queryParameters, FeatureLayer.SelectionMode.NEW);
+
+                // wait for the selection to finish
+                featureQueryResultListenableFuture.addDoneListener(() -> {
+                  // update the status text, enable the buttons and hide the progress indicator
+                  statusLabel.setText("Trace completed.");
+                  enableButtonInteraction();
+                  progressIndicator.setVisible(false);
+                  enableButtonInteraction();
+                });
+              }
+            });
+          }
+        } else {
+          statusLabel.setText("Trace failed.");
+          progressIndicator.setVisible(false);
+          new Alert(Alert.AlertType.ERROR, "Trace result not a utility element.").show();
+          enableButtonInteraction();
+        }
+
+      } catch (Exception e) {
+        statusLabel.setText("Trace failed.");
+        progressIndicator.setVisible(false);
+        // Note: this sample server may return a generic message in some circumstances when incompatible trace
+        // parameters are specified. This will be fixed in ArcGIS server 10.8
```

Comment:
is this a certain it will be fixed? I think it's probably ok not to state a time for when it will be fixed (the concern being it won't be fixed with 10.8 for whatever reason, or we forget to update the comment in the sample). Just stating that it may return a generic message is good enough for me.

Reply:
Good point, I've removed the reference to any upcoming fix